### PR TITLE
Refactor CodableRouter to ease code reuse

### PR DESCRIPTION
**Pull request by tunniclm**
## Description
Remove code duplication in the CodableRouter and expose CodableHelpers as building blocks for extensions to the CodableRouter.

## Motivation and Context
While working on a prototype codable authentication extension to the router I needed to write variations of the existing route registration functions from CodableRouter. These functions have a lot of code duplication to begin with, and with this new extension each function would need to be duplicated again to add the variations.

This motivated a refactor of these functions so they could be constructed from building block functions. These could then be exposed for use by router extensions.

## How Has This Been Tested?
Automated testing

## Checklist:

- [x] If applicable, I have updated the documentation accordingly.
- [x] If applicable, I have added tests to cover my changes.

### CODE REVIEW

**Comment by tunniclm** [comment]
@djones6 Good point! Done.

File: Sources/Kitura/CodableRouter.swift [hunk 1]
```diff
@@ -704,22 +402,258 @@ extension Router {
...
+     *                         closure completes execution.
+     * - Returns: The closure to pass to the codable route handler. The closure takes one argument
+     *            `(RequestError?)`.
+     *            If the argument is `nil` then the response will be considered successful, otherwise
+     *            it will be considered failed.
+     *
+     *            If successful, the HTTP status code will be set to `HTTPStatusCode.noContent` and no
+     *            body will be sent.
+     *
+     *            If failed, the HTTP status code used for the response wll be set to either the
```
Comment (djones6) [inline Sources/Kitura/CodableRouter.swift]: typo: wll

File: Sources/Kitura/CodableRouter.swift [hunk 2]
```diff
@@ -704,22 +402,258 @@ extension Router {
...
-        return (contentType.hasPrefix("application/json"))
+        return contentType.hasPrefix("application/json")
+    }
+
+    /**
+     * Get the HTTPStatusCode corresponding to the provided RequestError
+     *
+     * - Parameter from: The RequestError to map to a HTTPStatusCode
+     * - Returns: A HTTPStatusCode corresponding to the RequestError http code
+     *            if valid, or HTTPStatusCode.unknown otherwise
```
Comment (djones6) [inline Sources/Kitura/CodableRouter.swift]: Nitpick: Should the code elements here be back-ticked?

Comment (tunniclm) [reply Sources/Kitura/CodableRouter.swift]: Yes! I just missed them. :(

File: Sources/Kitura/CodableRouter.swift (hunk 1 above)
Comment (tunniclm) [reply Sources/Kitura/CodableRouter.swift]: :cry:

**Comment by djones6** [comment]
Thank you for adding this excellent documentation.  Let's get this merged and the (tiny) nitpicks can be addressed later.

**Pull request merged**